external/mmsegmentation/segmentation_tasks/apis/segmentation/model_wrappers/blur.py

```diff
@@ -78,18 +81,16 @@ def postprocess(self, outputs: Dict[str, np.ndarray], metadata: Dict[str, Any]):
             soft_threshold=self.soft_threshold,
             blur_strength=self.blur_strength
         )
-        hard_prediction = cv2.resize(hard_prediction, metadata['original_shape'][1::-1], 0, 0, interpolation=cv2.INTER_NEAREST)
-        
-        if 'feature_vector' not in outputs or 'saliency_map' not in outputs:
-            warnings.warn('Could not find Feature Vector and Saliency Map in OpenVINO output. '
-                'Please rerun OpenVINO export or retrain the model.')
-            metadata["saliency_map"] = None
+        hard_prediction = cv2.resize(hard_prediction, metadata['original_shape'][1::-1], 0, 0,
+                                     interpolation=cv2.INTER_NEAREST)
+        soft_prediction = cv2.resize(soft_prediction, metadata['original_shape'][1::-1], 0, 0,
+                                     interpolation=cv2.INTER_NEAREST)
+        metadata['soft_prediction'] = soft_prediction
```

Comment:
Maybe iterate for per-label soft prediction maps?

Reply:
`soft_prediction` includes soft prediction for each label as the tensor dimension is (H, W, Number of classes + 1). The iteration occurs here in `infer`: 
```
for label_index, label in self._label_dictionary.items():
      if label_index == 0:
          continue
      if len(soft_prediction.shape) == 3:
          current_label_soft_prediction = soft_prediction[:, :, label_index]
      else:
          current_label_soft_prediction = soft_prediction
      class_act_map = get_activation_map(current_label_soft_prediction)
      result_media = ResultMediaEntity(name=f'{label.name}',
                                       type='Soft Prediction',
                                       label=label,
                                       annotation_scene=dataset_item.annotation_scene,
                                       roi=dataset_item.roi,
                                       numpy=class_act_map)
      dataset_item.append_metadata_item(result_media, model=self.model)
```